### PR TITLE
feature: add hidden `--version-git` flag to show current git commit SHA-1

### PR DESCRIPTION
**Pull request by infinitewarp**
https://issues.redhat.com/browse/DISCOVERY-359

This change adds a new hidden `--version-git` argument to `qpc` that outputs either the current git commit SHA-1 or the value of the `QPC_COMMIT` environment variable, the latter of which may be useful in downstream image builds. The new argument mimics the standard `--version` argument behavior and exits immediately after displaying the value.

Example output:

```
❯ qpc --version
1.2.2

❯ qpc --version-git
7784740b4d977d81b33234ac4018ee939fb20388

❯ QPC_COMMIT=potato qpc --version-git
potato
```

Since this relies on finding the git repo, to verify that it isn't relying on the caller being in the project's root directory:

```
❯ pushd /tmp ; pwd ; qpc --version-git ; popd
/tmp ~/projects/qpc
/tmp
7784740b4d977d81b33234ac4018ee939fb20388
~/projects/qpc
```

New argument is omitted from the `--help` output:

```
❯ qpc --help
usage: qpc [-h] [--version] [-v] {server,cred,source,scan,report,insights} ...

positional arguments:
  {server,cred,source,scan,report,insights}

options:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  -v                    Verbose mode. Use up to -vvvv for more verbosity.
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ cryptography = ">=37.0.4"
 packaging = "^23.1"
 ruff = "^0.0.270"
```
Comment (bruno-fs) [inline pyproject.toml:21]: lol I just noticed now ruff is under normal dependencies.

File: qpc/cli.py [hunk 2]
```diff
@@ -75,8 +103,13 @@ def __init__(self, name="cli", usage=None, shortdesc=None, description=None):
         self.shortdesc = shortdesc
         if shortdesc is not None and description is None:
             description = shortdesc
-        self.parser = ArgumentParser(usage=usage, description=description)
+        self.parser = argparse.ArgumentParser(usage=usage, description=description)
         self.parser.add_argument("--version", action="version", version=VERSION)
+        self.parser.add_argument(
+            "--version-git",
+            action=VersionGitAction,
+            help=argparse.SUPPRESS,
```
Comment (infinitewarp) [inline qpc/cli.py:111]: This is the magic that hides the command from `--help`. Remove this arg, and it shows up normally.

**Review by abellotti — APPROVED** [review]

File: qpc/release.py [hunk 3]
```diff
@@ -9,3 +11,14 @@
...
 ENTRYPOINT = f"{QPC_VAR_PROGRAM_NAME}=qpc.__main__:main"
 URL = "https://github.com/quipucords/qpc"
+
+
+def get_current_sha1() -> str:
+    """Get the running code's current git commit SHA-1."""
+    if qpc_commit := os.environ.get("QPC_COMMIT", "").strip():
+        return qpc_commit
+    try:
+        repo = git.Repo(f"{__file__}/../..")
```
[the remark this answers is not in the document]
Comment (infinitewarp) [reply qpc/release.py:21]: I may actually switch to pathlib to be extra safe, though.

[the remark this answers is not in the document]
Comment (bruno-fs) [reply qpc/release.py:21]: I'm very tempted to enable a generic pathlib check on ruff if we all agree on preferring `Path`

[the remark this answers is not in the document]
Comment (bruno-fs) [reply qpc/release.py:21]: and, yeah, I always prefer pathlib, even though I agree this is cleaner then `Path(__file__).parent.parent`

File: qpc/cli.py (hunk 2 above)
Comment (abellotti) [reply qpc/cli.py:111]: 🪄

File: qpc/release.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (infinitewarp) [reply qpc/release.py]: I CONVERTED TO PATHLIB. PREACH THE GOSPEL OF PATHLIB. 😂

File: pyproject.toml (hunk 1 above)
Comment (bruno-fs) [reply pyproject.toml:21]: resolving since this is not introduced here

File: qpc/release.py [hunk 4]
```diff
@@ -9,3 +12,15 @@
...
 ENTRYPOINT = f"{QPC_VAR_PROGRAM_NAME}=qpc.__main__:main"
 URL = "https://github.com/quipucords/qpc"
+
+
+def get_current_sha1() -> str:
+    """Get the running code's current git commit SHA-1."""
+    if qpc_commit := os.environ.get("QPC_COMMIT", "").strip():
+        return qpc_commit
+    try:
+        repo_root = Path(__file__).absolute().parent.parent
```
Comment (bruno-fs) [inline qpc/release.py:22]: <3<3<3<3

**Review by bruno-fs — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (infinitewarp) [reply pyproject.toml:21]: https://github.com/quipucords/qpc/pull/261 nice

**Comment by infinitewarp** [comment]
Relates to JIRA: DISCOVERY-359

**Pull request merged**